st_files_connection/connection.py

```diff
@@ -0,0 +1,171 @@
+        **kwargs,
+    ) -> str:
+        pass
+
+    @overload
+    def read(
+        self,
+        path: str | Path,
+        input_format: Literal["csv"],
+        ttl: Optional[Union[float, int, timedelta]] = None,
+        **kwargs,
+    ) -> pd.DataFrame:
+        pass
+
+    @overload
+    def read(
+        self,
+        path: str | Path,
+        input_format: Literal["parquet"],
+        ttl: Optional[Union[float, int, timedelta]] = None,
+        **kwargs,
+    ) -> pd.DataFrame:
+        pass
+
+    def read(
+        self,
+        path: str | Path,
+        input_format: str = None,
+        ttl: Optional[Union[float, int, timedelta]] = None,
+        **kwargs,
+    ):
+        @cache_data(ttl=ttl, show_spinner="Running `files.read(...)`.")
+        def _read_text(path: str | Path, **kwargs) -> str:
+            if "connection_name" in kwargs:
+                kwargs.pop("connection_name")
+
+            with self.open(path, "rt", **kwargs) as f:
+                return f.read()
+
+        @cache_data(ttl=ttl, show_spinner="Running `files.read(...)`.")
+        def _read_csv(path: str | Path, **kwargs) -> pd.DataFrame:
+            if "connection_name" in kwargs:
+                kwargs.pop("connection_name")
+
+            with self.open(path, "rt") as f:
+                return pd.read_csv(f, **kwargs)
+
+        @cache_data(ttl=ttl, show_spinner="Running `files.read(...)`.")
+        def _read_parquet(path: str | Path, **kwargs) -> pd.DataFrame:
+            if "connection_name" in kwargs:
+                kwargs.pop("connection_name")
+
+            with self.open(path, "rb") as f:
+                return pd.read_parquet(f, **kwargs)
+        
+        if input_format == 'text':
+            return _read_text(path, connection_name=self._connection_name, **kwargs)
+        elif input_format == 'csv':
+            return _read_csv(path, connection_name=self._connection_name, **kwargs)
+        elif input_format == 'parquet':
+            return _read_parquet(path, connection_name=self._connection_name, **kwargs)
+        # TODO: if input_format is None, try to infer it from file extension
+        raise ValueError(f"{input_format} is not a valid value for `input_format=`.")
+
+    def _repr_html_(self) -> str:
+        module_name = getattr(self, "__module__", None)
+        class_name = type(self).__name__
+        if self._connection_name:
+            name = f"`{self._connection_name}` "
+            if len(self._secrets) > 0:
+                cfg = f"""- Configured from `[connections.{self._connection_name}]`
+                """
+            else:
+                cfg = ""
+        else:
+            name = ""
+            cfg = ""
+        md = f"""
+        ---
+        **st.connection {name}built from `{module_name}.{class_name}`**
+        {cfg}
+        - Protocol: `{self.protocol}`
+        - Learn more using `st.help()`
+        ---
+        """
+        return md
```

Comment:
This can be simplified to look more like the more concise (but equivalent) implementation in `ExperimentalBaseConnection` (which is in part shorter because it assumes `self._connection_name` is always set, which I think will always be the case now that name is the first param).

I'm also slightly worried about this method because both `SQLConnection` and this connection seem to be essentially duplicating the default implementation just to get a line in between the "Configured from" and "Learn more..." lines. I wonder if reordering/rewriting things slightly would make sense so that we can get some code reuse for connections that want something that's very close to the default implementation but only slightly tweaked. Probably something to think about before GA

Reply:
Agreed, maybe an attribute or a property that gets checked for this extra stuff in the base class?